### PR TITLE
Added Create effect from Result. 

### PR DESCRIPTION
**Pull request by Lanayx**

### CODE REVIEW

**Comment by JohSand** [comment]
Hi, thanks for your interest in this project.

Being able to create effects from a result is a nice addition. However, those variables are there in order to get the overload resolution to work properly. 

I admit it is a bit hacky, and a better solution could be using the extension method trick that the builder is using.

File: src/Orsak/Effect.fs [hunk 1]
```diff
@@ -343,7 +343,7 @@ type Effect =
     ///         Orsak.Effect.Create(fun (e: #MyEffectProvider) -> e.Effect.DoEffect(x))
     ///   </code>
     /// </example>
-    static member Create(f: 'a -> Task<'b>, [<Optional>] _medium: byte) : Effect<'a, 'b, _> =
+    static member Create(f: 'a -> Task<'b>) : Effect<'a, 'b, _> =
```
Comment (JohSand) [inline src/Orsak/Effect.fs]: We need that parameter for overload resolution, or we need to create a module with a type extension in order for the overload resolution to work

File: src/Orsak/Effect.fs [hunk 2]
```diff
@@ -353,9 +353,16 @@ type Effect =
     /// <summary>
     /// Creates an effect from a function.
     /// </summary>
-    static member Create(f: 'a -> ValueTask<'b>, [<Optional>] _medium: byte) : Effect<'a, 'b, _> =
+    static member Create(f: 'a -> ValueTask<'b>) : Effect<'a, 'b, _> =
```
Comment (JohSand) [inline src/Orsak/Effect.fs]: Same as above

**Review by JohSand — CHANGES_REQUESTED** [review]

**Comment by Lanayx** [comment]
Thank you for the clarifications, I've updated the code

**Pull request merged**